BeRich/Network/TradingDataNetworkFetcher.swift

```diff
@@ -2,33 +2,24 @@
 import Foundation
 
 protocol TradingDataNetworkFetching {
-    func getTickers(completion: @escaping (BinanceTikers?) -> Void)
+    func getTickers() async -> BinanceTikers?
 }
 
 final class TradingDataNetworkFetcher: TradingDataNetworkFetching, ObservableObject {
-    private let request: NetworkRequest
-
-    init(request: @escaping NetworkRequest) {
-        self.request = request
-    }
-
-    func getTickers(completion: @escaping (BinanceTikers?) -> Void) {
-        guard let url = BinanceApi.Method.exchangeInfo.url() else { return completion(nil) }
-        request(url) { data in
-            switch data {
-            case let .success(data):
-                do {
-                    let binanceTickers = try decodeJSON(type: BinanceTikers.self, from: data)
-                    completion(binanceTickers)
-                } catch {
-                    print(error)
-                    completion(nil)
-                }
-            case let .failure(error):
-                print(error)
-                completion(nil)
-            }
+    func getTickers() async -> BinanceTikers? {
+        guard let url = BinanceApi.Method.exchangeInfo.url() else { return nil }
```

Comment:
```suggestion
        guard let url = BinanceApi.Method.exchangeInfo.url() else {
          assertionFailure()
          return nil
        }
```
Ассерты помогут детектить неожиданное поведение кода во внутренних билдах (в релизном билде они вырежутся)

Reply:
Fixed